atlasdb-commons/src/main/java/com/palantir/async/initializer/AsyncInitializer.java

```diff
@@ -102,4 +103,8 @@ default void assertNotInitialized() {
     void tryInitialize();
 
     void cleanUpOnInitFailure();
+
+    default void close() {
+        // you must implement this method if you are asynchronously initializing a closeable class
```

Comment:
BLUF: let's not make this default, and instead add a helpful javadoc

It seems to me that there are two error cases here:
a) Having this default, and accidentally not implementing it for a closable class
b) Having this not default, and accidentally implementing it for (or having a compile error) a non-closable class.

b) seems rather trivial - just make an empty impl if the class has nothing to do. Falling foul of (a) could be harmful - we might forget to close something and cause a memory leak, or something else icky.

Therefore, I'd prefer to not make this default, and instead to javadoc that this is important for closable classes, but can be empty for non-closable classes.

Reply:
That's fair, also javadocs incoming

Reply:
We're implementing a better architecture for this on Part 4. For now, let's leave it as is.